templates/model/spec.go

```diff
@@ -224,6 +225,10 @@ func (s *Step) UnmarshalYAML(n *yaml.Node) error {
 		s.StringReplace = new(StringReplace)
 		unmarshalInto = s.StringReplace
 		s.StringReplace.Pos = s.Pos
+	case "append":
```

Comment:
I wonder if "append" is too generic of a name. Even within this PR, I think we could have two functions:

- `append_file`
- `append_string`

Reply:
Is there a use for append_string that couldn't be done using the template language instead?

Reply:
Thinking more about it, is it worth having an append? The regex_replace action as it currently stands could do the exact same thing. A simple test I made:

```
		{
			name: "append",
			initContents: map[string]string{
				"a.txt": "apple banana\nbanana apple\napple apple\n", //nolint:dupword
			},
			rr: &model.RegexReplace{
				Paths: modelStrings([]string{"."}),
				Replacements: []*model.RegexReplaceEntry{
					{
						Regex: model.String{Val: "$"},
						With:  model.String{Val: "foobar\n"},
					},
				},
			},
			want: map[string]string{
				"a.txt": "apple banana\nbanana apple\napple apple\nfoobar\n", //nolint:dupword
			},
		},
```

Reply:
My proposal is that we can have a single `append` action that supports with `with` and `with_file`. We can wait to build the `with_file` option until somebody needs it.

Reply:
What would the `with_file` do, append one file onto another?

Reply:
I agree it's possible with regex, but I don't think it's a good devex. I like Dave's proposal:

```yaml
- action: 'append'
  with: 'dfjkdladf'

- action: 'append'
  with_file: './my-file.txt'
```

that would add `dfjkdladf` and `$(cat ./my-file.txt)` to the template.

Reply:
> What would the with_file do, append one file onto another?

Yeah, that was my thought. It won't be a common use case, but suppose you want to append a huge amount of data to a file; rather than inlining the huge amount of data in the spec.yaml you could store it as append_me.txt in git then use `append` `with_file` to tack it onto the end of an existing file.

I think we shouldn't build this now though; we may never need it.

Reply:
OK, I'll leave `with_file` out for now.